packages/components/switch/src/switch.vue

```diff
@@ -100,162 +100,150 @@ import {
   useSize,
 } from '@element-plus/hooks'
 import { switchEmits, switchProps } from './switch'
-
 import type { CSSProperties } from 'vue'
 
 const COMPONENT_NAME = 'ElSwitch'
 
-export default defineComponent({
-  name: COMPONENT_NAME,
-  components: { ElIcon, Loading },
-
-  props: switchProps,
-  emits: switchEmits,
-
-  setup(props, { emit }) {
-    const { formItem } = useFormItem()
-    const switchDisabled = useDisabled(computed(() => props.loading))
-    const ns = useNamespace('switch')
-
-    const { inputId } = useFormItemInputId(props, {
-      formItemContext: formItem,
-    })
-
-    const switchSize = useSize()
-    const isModelValue = ref(props.modelValue !== false)
-    const input = ref<HTMLInputElement>()
-    const core = ref<HTMLSpanElement>()
-
-    const switchKls = computed(() => [
-      ns.b(),
-      ns.m(switchSize.value),
-      ns.is('disabled', switchDisabled.value),
-      ns.is('checked', checked.value),
-    ])
+defineOptions({
+  name: 'ElSwitch',
+})
 
-    const coreStyle = computed<CSSProperties>(() => ({
-      width: addUnit(props.width),
-    }))
+const props = defineProps(switchProps)
+const emit = defineEmits(switchEmits)
 
-    watch(
-      () => props.modelValue,
-      () => {
-        isModelValue.value = true
-      }
-    )
+const { formItem } = useFormItem()
+const ns = useNamespace('switch')
 
-    watch(
-      () => props.value,
-      () => {
-        isModelValue.value = false
-      }
-    )
+const { inputId } = useFormItemInputId(props, {
+  formItemContext: formItem,
+})
 
-    const actualValue = computed(() => {
-      return isModelValue.value ? props.modelValue : props.value
-    })
+const switchSize = useSize()
+const switchDisabled = useDisabled(computed(() => props.loading))
+const isModelValue = ref(props.modelValue !== false)
```

Comment:
**Suggestion:** maybe we should rename this to `isControlled`, and the implementation is:

```suggestion
const isModelValue = ref(isBoolean(props.modelValue))
```